packages/block-library/src/buttons/deprecated.js

```diff
@@ -1,9 +1,79 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
 /**
  * WordPress dependencies
  */
-import { InnerBlocks } from '@wordpress/block-editor';
+import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';
+
+/**
+ * @param {Object} attributes Block's attributes.
+ */
+const migrateWithLayout = ( attributes ) => {
+	if ( !! attributes.layout ) {
+		return attributes;
+	}
+
+	const { contentJustification, orientation } = attributes;
```

Comment:
I'm not sure I understand. These are used here to migrate any legacy settings to the new layout, e.g. if there's a deprecated block with "orientation" set we use that value in layout when we update the block. Or is there another way of doing it?

In the fixtures these attributes only appear in the deprecated versions of Buttons. That should be expected, right?

Reply:
Oh wait, you mean I should be removing them from the updated attributes?

Reply:
> I should be removing them from the updated attributes?

Yes. We extract them here and use them for the `layout` but since the current version doesn't contain them, these attributes shouldn't exist at the final migrated version of the block.